### PR TITLE
Sensing Component Integration

### PR DESCRIPTION
**Pull request by DennisSSDev**
I have added the component, but it's not entirely complete as the sensor actor is not yet completed. 
We need to setup a separate task, but I believe we can at least do the code review for the sensing component for now.

Also, there is a bit of a follow-up fix for the find target with axis changes so it would also be great to take a look at that!

### CODE REVIEW

**Comment by louisgv** [comment]
Is this a continuation of #3 ? If so, we should change target to focus the diff.

**Comment by DennisSSDev** [comment]
> Is this a continuation of #3 ? If so, we should change target to focus the diff.

Not in particular, I think I accidentally added the fix on this branch from the suggestions on #3 . could repeat e1915ce on #3 just to be in sync

**Comment by louisgv** [comment]
@DennisSSDev I changed the base of the merge. So now we only see the diff between the new change you made and the change you already incorporated in #3 (this is what I meant by change target)

File: Source/NujasCombat/Private/TargetSensingComponent.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+void UTargetSensingComponent::AddEnemy(AActor* Enemy)
+{
+}
+
+void UTargetSensingComponent::RemoveEnemy(AActor* Enemy)
+{
+}
+
+AActor * UTargetSensingComponent::SearchForClosestEnemy()
```
Comment (ico3939) [inline Source/NujasCombat/Private/TargetSensingComponent.cpp]: Wouldn't this function and `GetClosestEnemy()` basically accomplish the same task?

**Pull request merged**